src/component.rs

```diff
@@ -81,25 +79,25 @@ impl Deref for TrailerSeparator {
         match self {
             TrailerSeparator::ColonSpace => ": ",
             TrailerSeparator::SpacePound => " #",
+            TrailerSeparator::__NonExhaustive => "",
         }
     }
 }
 
 impl fmt::Display for TrailerSeparator {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            TrailerSeparator::ColonSpace => f.write_str(": "),
-            TrailerSeparator::SpacePound => f.write_str(" #"),
-        }
+        f.write_str(self)
     }
 }
 
-impl From<&str> for TrailerSeparator {
-    fn from(sep: &str) -> Self {
+impl std::str::FromStr for TrailerSeparator {
```

Comment:
nit: I prefer to keep inline nested paths to a minimum for easier readability (which also means _not_ importing `std::fmt::Result`, but using `fmt::Result` instead to prevent confusion; not that you did this, but just to expand on my reasoning).

I've extracted this one (and others) to `use` statements at the top of the file.

Reply:
For me, it is a question of how often the type is used.  If it is rarely used, then I just use the full path rather than `use`.

Reply:
Right. I prefer having all imports at the top and "out of the way" before digging into the implementation logic itself to reduce the visual noise. But I realise others might have different opinions on matters like these.